ducktape/command_line/main.py

```diff
@@ -56,6 +57,7 @@ def parse_args():
                              "This is primarily useful for test developers who want to interact with running " +
                              "services after a test has run. ")
     parser.add_argument("--version", action="store_true", help="display version")
+    parser.add_argument("--inject", action="store", help="Inject these arguments into the specified test. Use JSON to format.")
```

Comment:
I like the idea of being able to copypaste a one-line command from the log to target a particular test, but that could be implemented with either approach.

I actually have a strong preference for the former approach because it is better suited for exploratory testing. It can even be used on methods without parametrization: for example,
```
def test_something(self, x=-1, y=2):
   ...test logic...
```
```
ducktape test_something --args "{'x': 100, 'y': 15}"
```